src/wallet/mod.rs

```diff
@@ -100,6 +121,45 @@ impl WalletManager {
             .to_owned())
     }
 
+    pub async fn apply_keystream(&self, password: &str, mut data: Vec<u8>) -> anyhow::Result<Vec<u8>> {
+        let mut hasher = VarBlake2b::new(16).unwrap();
+        hasher.update(b"nonce");
```

Comment:
The nouce is for the `Aes128Ctr` cipher, but maybe we can hash the salt of `pbkdf2` with `crate::encoding::blake2b` and use it as the nonce? I don't know if that is a good practice.

Reply:
All good. Let's leave it for now and we can tweak it later. Although this might cause problems with older wallets if we change it later. We would have to have an upgrade process... Maybe we can include a version number in the wallet? Maybe something like:
```rust
struct WalletStorage {
  version: usize
  storage: WalletType
}

enum {
  Clear { wallets: HashMap<> },
  Encrypted { salt: String, encrypted: String }
}
```
What do you think?

Reply:
What if we use this crate [age](https://docs.rs/age/0.5.1/age/) for encrypting the file instead? Can simplify a lot the implementation.

Reply:
SGTM!